plugin/src/main/kotlin/com/varabyte/kobweb/intellij/services/project/KobwebProjectCacheService.kt

```diff
@@ -12,36 +15,57 @@ import java.util.*
  * waste time processing code that isn't relevant to Kobweb (e.g. if we're in a multiplatform project with Kobweb,
  * Android, and other types of modules).
  *
+ * Note there's also an option to mark an element as explicitly NOT being part of a Kobweb project. This allows early
+ * aborting on followup checks, allowing the caller to distinguish the difference between "get returns null because
+ * this is not a kobweb project" vs "get returns null because we haven't put the value in the cache yet".
+ *
  * @see KobwebProject
  */
 interface KobwebProjectCacheService : Iterable<KobwebProject> {
     operator fun get(module: Module): KobwebProject?
-    operator fun get(jar: VirtualFile): KobwebProject?
+    operator fun get(klib: VirtualFile): KobwebProject?
+    fun isNotKobweb(module: Module) = get(module) == null
+    fun isNotKobweb(klib: VirtualFile) = get(klib) == null
+
     fun add(project: KobwebProject)
     fun addAll(collection: Collection<KobwebProject>)
-    fun removeIf(filter: (KobwebProject) -> Boolean)
+    fun markNonKobweb(element: PsiElement)
```

Comment:
It's actually correct; however, my implementation was bad and there was also some incorrect code in the base interface which I probably autogenerated when I was tired.

I did add a clarifying comment though to explain my reasoning.

Reply:
I see. Now the issue is reversed though, 2 of the methods accept a PsiElement and one accepts a module or virtualfile. Shouldnt you then also make `get` accept a PsiElement?

Reply:
My thinking here is the things that can NOT be Kobweb projects are far more flexible than the things that can be Kobweb projects, which is why the get calls are more limited.

Looking at the code in KobwebProjectUtils.kt, if we made the "get" call take a PsiElement, then we'd need to duplicate the logic around fetching the Gradle module for example.

Reply:
Hm in my opinion only allowing PSI elements restricts us even more. What if we dont have a PsiElement available? Also what would someone ever want to mark as non-kobweb other than a module or file? What do you think about having the cache service accept a Module or file as core its core api and adding extension functions on the cache service that handle a PsiElement?

Reply:
It's hard for me to answer until we encounter non PsiElement situations. In my experience, so many extension points ask you a question using a PsiElement as the main context, so that's what we're supporting at the moment. If in the future we have some additional case to handle we can update the caching API.

What do you think about committing this as is (the code works, it's not particularly complex) and then you can propose a followup commit with your extension function idea?

Reply:
Sure. Its not really urgent right now, and if we ever need to support more approaches, they can be added without affecting usages